### PR TITLE
cmd/scollector: support auth token

### PR DESCRIPTION
**Pull request by kylebrandt**

### CODE REVIEW

File: cmd/scollector/main.go [hunk 1]
```diff
@@ -50,14 +50,18 @@ var (
...
+	XAccessToken string
 	http.RoundTripper
 }
 
 func (t *scollectorHTTPTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	if req.Header.Get("User-Agent") == "" {
 		req.Header.Add("User-Agent", t.UserAgent)
 	}
+	if t.XAccessToken != "" {
+		req.Header.Add("X-Access-Token", t.XAccessToken)
```
Comment (captncraig) [inline cmd/scollector/main.go]: We already have an AuthToken in the collect package. Should be as easy as setting that from the scollector config.

https://github.com/bosun-monitor/bosun/blob/master/collect/collect.go#L63

File: cmd/scollector/conf/conf.go [hunk 2]
```diff
@@ -50,6 +50,10 @@ type Conf struct {
 	// UseNtlm specifies if HTTP requests should authenticate with NTLM.
 	UseNtlm bool
 
+	// XAccessToken is an optional string that sets the X-Access-Token HTTP header
+	// which is used to authenticate against Bosun
+	XAccessToken string
```
Comment (captncraig) [inline cmd/scollector/conf/conf.go]: Lets call this `AuthToken` perhaps? Looks like maybe we've been a tad inconsistent.

File: cmd/scollector/main.go (hunk 1 above)
Comment (kylebrandt) [inline cmd/scollector/main.go]: Should I follow the same pattern in the metadata package? Create a global var?

Comment (captncraig) [inline cmd/scollector/main.go]: yeah. It may be there already too.

**Pull request merged**